packages/devtools_app/lib/src/debugger/codeview.dart

```diff
@@ -264,6 +273,11 @@ class _CodeViewState extends State<CodeView>
             child: Expanded(
               child: Scrollbar(
                 controller: textController,
+                // Only listen for vertical scroll notifications (ignore those
+                // from the nested horizontal SingleChildScrollView):
+                notificationPredicate: (ScrollNotification notification) =>
```

Comment:
We should use `notification.depth` to predicate, otherwise, all the descendants of Scrollables will interfere.

Reply:
Thanks, changed to use `notificationDepth`.

Reply:
are we sure this notificationPredicate is needed? If it is needed, there are likely multiple other places in the UI where we need notificationPredicates. I thought it wasn't needed as we explicitly specify the scrollController.

Reply:
The `notificationPredicate` is needed, otherwise the vertical scrollbar responds to horizontal scroll events (and shows up at the bottom along with the horizontal scrollbar, see image). I'm guessing scrollbars in other parts of the UI would be similarly broken if this was needed for them?

![image](https://user-images.githubusercontent.com/21270878/129286214-ec4990c2-47ca-49f5-89e3-0a594ca196ac.png)

Reply:
Hi @xu-baolin! I see you added the [`notificationPredicate`](https://github.com/flutter/flutter/commit/e7772d0e2ecb9529f692556e8b446de65219372c) 😄  Could you clarify what the purpose of the `notificationPredicate` is vs. the `scrollController`? Is the `scrollController` no longer used? Thanks!

Reply:
Ok I will check that.

Reply:
Oh this may already be fixed, we noticed this recently in https://github.com/flutter/flutter/issues/87697.
The scrollbar uses the scroll controller but also listens and responds to scroll events. This is useful like in this 2D case where the scrollbars might want to know about both axes and respond when either one scrolls. They just aren't behaving correctly here, should be fixed by https://github.com/flutter/flutter/pull/87698.